### DISCUSSION TITLE
missing transform function while migrating from request-response

### DISCUSSION THREAD

**Discussion by ehmad11** [sindresorhus/got#1791] [Q&A] [answered]
#### What would you like to discuss?

Hello,

I am replacing request-promise with got in my package:
https://github.com/ehmad11/netsuite-rest/blob/master/netsuite-rest.js

I was using transform method of [request-method](https://www.npmjs.com/package/request-promise) which takes a function to transform the response into a custom value with which the promise is resolved.

is there any way I can _transform_ in 'got'?

my current _transform_ code:

            transform: (body, response) => {
                let data = {}
                if (body)
                    data = JSON.parse(body)
                return {
                    statusCode: response.statusCode,
                    'headers': response.headers,
                    'data': data
                };
            },

#### Checklist

- [ ✓] I have read the documentation.

**Comment by sindresorhus** [comment, marked as answer]
You could use the `afterResponse` hook.
